include/swift/ABI/MetadataValues.h

```diff
@@ -997,8 +997,10 @@ enum class ExclusivityFlags : uintptr_t {
   // Read or Modify).
   ActionMask       = 0x1,
 
-  // Downgrade exclusivity failures to a warning.
+#ifndef NDEBUG
```

Comment:
One possibility to remove the need to expose this in the ABI header is to separate out the unit tests that abort into a separate testing tool (similar to swift-ide-test) and turn the unit tests into file check tests.